telegram/_telegramobject.py

```diff
@@ -177,14 +203,38 @@ def de_json(cls: Type[TO_co], data: Optional[JSONDict], bot: "Bot") -> Optional[
             The Telegram object.
 
         """
-        data = cls._parse_data(data)
+        return cls._de_json(data=data, bot=bot)
+
+    @classmethod
+    def _de_json(
+        cls: Type[TO_co], data: Optional[JSONDict], bot: "Bot", api_kwargs: JSONDict = None
+    ) -> Optional[TO_co]:
+        if cls.__INIT_PARAMS is None:
+            signature = inspect.signature(cls)
+            cls.__INIT_PARAMS = set(signature.parameters.keys())
 
         if data is None:
             return None
 
-        if cls == TelegramObject:
-            return cls()
-        return cls(bot=bot, **data)
+        api_kwargs = api_kwargs or {}
+
+        # try-except is significantly faster in case we already have a correct argument set
+        try:
+            obj = cls(**data, api_kwargs=api_kwargs)
+        except TypeError as exc:
+            if "__init__() got an unexpected keyword argument" in str(exc):
+                kwargs: JSONDict = {}
```

Comment:
I had a one line along the lines `(kwargs if key in __INIT_PARAMS else api_kwargs)[key] = value)` but pylint & mypy made that infeasible …

What solution using set difference did you have in mind?

I'd like to avoid having to call `_parse_data` to save one copying of the data. BTW, one could also add tests to all `de_json` tests that ensure that `data` is not changed. That could be done in a standalone PR though IMO b/c a) that behavior is not new in the PR b) I'm also fine with keeping that an implementation detail.

Reply:
edit: I actually found out that just checking `if __INIT_PARAMS is None` doesn't work and with the fix I was able to use the on-liner after all …

Reply:
nvm the set difference, I thought we can take set diff of `set(data)` and `__INIT_PARAMS`, which would give us keys to put in `api_kwargs`, which can be done by pop()ing from `data`. And then finally pass `api_kwargs` and `data` directly as `cls(api_kwargs=api_kwargs, **data)`.

But I don't know if this will be faster or not.

Reply:
that would require one pass through data for `set(data)` and another one b/c we have to call `_parse_data` … theoretically that should be a bit slower, i guess …